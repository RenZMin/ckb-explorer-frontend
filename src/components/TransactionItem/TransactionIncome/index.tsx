import BigNumber from 'bignumber.js'
import { TransactionIncomePanel, TransactionCapacityValuePanel } from './styled'
import { shannonToCkb } from '../../../utils/util'
import { localeNumberString } from '../../../utils/number'
import DecimalCapacity from '../../DecimalCapacity'

export default ({ income }: { income: string }) => {
  let bigIncome = new BigNumber(income)
  if (bigIncome.isNaN()) {
    bigIncome = new BigNumber(0)
  }
  return (
    <TransactionIncomePanel>
      <TransactionCapacityValuePanel increased={bigIncome.isGreaterThanOrEqualTo(0)}>
        <DecimalCapacity
          value={`${bigIncome.isPositive() ? '+' : ''}${localeNumberString(shannonToCkb(bigIncome))}`}
          color="inherit"
        />
      </TransactionCapacityValuePanel>
    </TransactionIncomePanel>
  )
}
