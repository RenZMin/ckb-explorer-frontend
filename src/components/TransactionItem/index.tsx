import React, { ReactNode } from 'react'
import { Link } from 'react-router-dom'
import RightArrowIcon from '../../assets/input_arrow_output.png'
import DownArrowIcon from '../../assets/input_arrow_output_down.png'
import { parseDate } from '../../utils/date'
import { localeNumberString } from '../../utils/number'
import { isMobile, isScreenSmallerThan1200 } from '../../utils/screen'
import { adaptPCEllipsis, adaptMobileEllipsis } from '../../utils/string'
import TransactionCell from './TransactionItemCell'
import TransactionCellList from './TransactionItemCellList'
import TransactionIncome from './TransactionIncome'
import { FullPanel, TransactionHashBlockPanel, TransactionCellPanel, TransactionPanel } from './styled'
import i18n from '../../utils/i18n'
import { CellType } from '../../utils/const'

export interface CircleCorner {
  top?: boolean
  bottom?: boolean
}

const TransactionItem = ({
  transaction,
  address,
  isBlock = false,
  titleCard,
  circleCorner = {
    top: false,
    bottom: false,
  },
}: {
  transaction: State.Transaction
  address?: string
  isBlock?: boolean
  titleCard?: ReactNode | null
  circleCorner?: CircleCorner
}) => {
  const txHashMobile = adaptMobileEllipsis(transaction.transactionHash, 10)
  const txHashPC = adaptPCEllipsis(transaction.transactionHash, 17, 28)

  return (
    <TransactionPanel id={isBlock && transaction.isCellbase ? 'cellbase' : ''} circleCorner={circleCorner}>
      {titleCard}
      <TransactionHashBlockPanel>
        <div className="transaction_item__content">
          <Link to={`/transaction/${transaction.transactionHash}`}>
            <span className="transaction_item__hash monospace">{isMobile() ? txHashMobile : txHashPC}</span>
          </Link>
          {!isBlock && (
            <div className="transaction_item__block">
              {`(${i18n.t('block.block')} ${localeNumberString(transaction.blockNumber)})  ${parseDate(
                transaction.blockTimestamp,
              )}`}
            </div>
          )}
        </div>
      </TransactionHashBlockPanel>
      <TransactionCellPanel>
        <div className="transaction_item__input">
          <TransactionCellList
            cells={transaction.displayInputs}
            transaction={transaction}
            render={cell => <TransactionCell cell={cell} address={address} cellType={CellType.Input} key={cell.id} />}
          />
        </div>
        <img src={isScreenSmallerThan1200() ? DownArrowIcon : RightArrowIcon} alt="input and output" />
        <div className="transaction_item__output">
          {transaction.displayOutputs && transaction.displayOutputs.length !== 0 ? (
            <TransactionCellList
              cells={transaction.displayOutputs}
              transaction={transaction}
              render={cell => (
                <FullPanel key={cell.id}>
                  <TransactionCell cell={cell} address={address} cellType={CellType.Output} />
                </FullPanel>
              )}
            />
          ) : (
            <div className="transaction_item__output__empty">{i18n.t('transaction.empty_output')}</div>
          )}
        </div>
      </TransactionCellPanel>
      {address && <TransactionIncome income={transaction.income} />}
    </TransactionPanel>
  )
}

export default TransactionItem
