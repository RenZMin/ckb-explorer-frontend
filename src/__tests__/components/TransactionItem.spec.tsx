import { lement } from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import * as timezoneMock from 'timezone-mock'
import { BrowserRouter } from 'react-router-dom'
import TransactionItem from '../../components/TransactionItem'
import { toCamelcase } from '../../utils/util'

const tx = {
  is_cellbase: false,
  witnesses: [
    '0x550000001000000055000000550000004100000074032cdce1e2a28f4a45c37578656fc6ed98762f0ae276a668e5df6351336da56cd16075998153bd08b0164e3d01c74bb0c8e998b28efe619b3db71dfe13e0b501',
    '0x5500000010000000550000005500000041000000bbb28d6722caff9fcd6f0b4d11ae78e858a411eb846620163d659d7e6fc1af195ea68c8e8f4b196f6ce5498b21d0132f17256d4593afbda64895e68b06280cae01',
    '0x',
    '0x',
  ],
  cell_deps: [
    {
      dep_type: 'dep_group',
      out_point: {
        index: 0,
        tx_hash: '0x71a7ba8fc96349fea0ed3a5c47992e3b4084b031a42264a018e0072e8172e46c',
      },
    },
    {
      dep_type: 'code',
      out_point: {
        index: 2,
        tx_hash: '0xe2fb199810d49a4d8beec56718ba2593b665db9d52299a0f9e6e75416d73ff5c',
      },
    },
  ],
  header_deps: null,
  transaction_hash: '0x89699abf948440ce172a6ef118cbf6986aa7c80d24d57d9309ca11929cebe5c4',
  transaction_fee: '4818',
  block_number: '1542388',
  version: '0',
  block_timestamp: '1587648896878',
  display_inputs: [
    {
      id: '2045056',
      from_cellbase: false,
      capacity: '10023577190.0',
      address_hash: 'ckb1qyqdpmfusmag3m9qjz3fenwa3qym3nc8prsqxxggjg',
      generated_tx_hash: '0xbebc637dd1eee18e73590683478b6038b75427a6d3557d90e34b9a0b44c7c2cc',
      cell_index: '1',
      cell_type: 'normal',
    },
    {
      id: '2076355',
      from_cellbase: false,
      capacity: '190296799086.0',
      address_hash: 'ckb1qyq93m8fdcx2knfmjyyn4q69zyhy7ggg835sl03lwy',
      generated_tx_hash: '0x7d5b5854a6d9a8dfb33188463d51d8c2d03a6b160815667cf15101f7b180f44b',
      cell_index: '75',
      cell_type: 'normal',
    },
    {
      id: '2051475',
      from_cellbase: false,
      capacity: '200221141693.0',
      address_hash: 'ckb1qyq93m8fdcx2knfmjyyn4q69zyhy7ggg835sl03lwy',
      generated_tx_hash: '0x49053d8717a41115a0c797a42abb3a4880ea2cdfbb8458f773b651b02737959c',
      cell_index: '62',
      cell_type: 'normal',
    },
    {
      id: '2063865',
      from_cellbase: false,
      capacity: '203966621540.0',
      address_hash: 'ckb1qyq93m8fdcx2knfmjyyn4q69zyhy7ggg835sl03lwy',
      generated_tx_hash: '0x59d72c3e3a5412a5e90eb1466eb801cdfaa6ccfa180d84a594c88fd426945c11',
      cell_index: '79',
      cell_type: 'normal',
    },
  ],
  display_outputs: [
    {
      id: '2082161',
      capacity: '590000000000.0',
      address_hash: 'ckb1qyqwr7vnfh6pcw2az25kkz3f4eq5a7ggec3qjyrfah',
      status: 'live',
      consumed_tx_hash: '',
      cell_type: 'nervos_dao_deposit',
    },
    {
      id: '2082162',
      capacity: '14508134691.0',
      address_hash: 'ckb1qyq907p79d6kkz7kxgu0awea8esvafm0586sh3kk9v',
      status: 'live',
      consumed_tx_hash: '',
      cell_type: 'normal',
    },
  ],
  income: null,
}

describe('TransactionItem Component', () => {
  let component: ReactElement

  beforeAll(() => {
    timezoneMock.register('UTC')

    const transaction = toCamelcase(tx) as State.Transaction
    component = (
      <BrowserRouter>
        <TransactionItem transaction={transaction} />
      </BrowserRouter>
    )
  })

  it('shallow renders', () => {
    const wrapper = renderer.create(component).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('Component Render', () => {
    const wrapper = shallow(component)
    expect(wrapper).toBeDefined()
    expect(wrapper.find(TransactionItem)).toHaveLength(1)
  })
})
