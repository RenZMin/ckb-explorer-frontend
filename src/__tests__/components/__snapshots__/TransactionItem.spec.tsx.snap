// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionItem Component shallow renders 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 16px;
}

.c4 .transaction__cell_address {
  color: undefined;
  font-weight: 500;
  height: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 .transaction__cell_address a {
  color: undefined;
}

.c4 .transaction__cell_address a:hover {
  color: undefined;
}

.c6 {
  color: #000000;
  margin-left: 15px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-height: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 > span {
  margin-left: 5px;
}

.c6 .transaction__cell__without__icon {
  margin-right: 21px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 2px;
}

.c12 img {
  margin-left: 5px;
  width: 16px;
  height: auto;
  cursor: pointer;
}

.c5 {
  width: 16px;
  height: auto;
  margin: 0px 5px 3px 0;
}

.c11 {
  width: 15px;
  height: 15px;
  height: auto;
  margin: 1px 0 0 7px;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c7 .decimal__zeros {
  margin-bottom: 1px;
}

.c7 .decimal__unit {
  margin-left: 5px;
}

.c8 {
  margin-bottom: 1px;
  font-size: 12px;
  color: #999999;
}

.c9 {
  margin-bottom: 1px;
  font-size: 12px;
  color: #999999;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: space-between;
  -webkit-box-align: space-between;
  -ms-flex-align: space-between;
  align-items: space-between;
  width: 100%;
}

.c3 .transaction_item__view_all {
  font-size: 16px;
  margin-top: 20px;
  height: 20px;
}

.c0 {
  width: 100%;
  margin-top: 4px;
  border-radius: 0 0 0 0;
  box-shadow: 2px 2px 6px 0 #dfdfdf;
  background-color: #ffffff;
  padding: 10px 40px 15px 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 16px;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 10px;
}

.c1 .transaction_item__content {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 10px;
}

.c1 .transaction_item__content .transaction_item__hash {
  font-size: 14px;
  font-weight: 500;
}

.c1 .transaction_item__content .transaction_item__block {
  color: #000000;
}

.c1:after {
  content: '';
  background: #e2e2e2;
  height: 1px;
  width: 100%;
  display: block;
  -webkit-transform: scaleY(1);
  -ms-transform: scaleY(1);
  transform: scaleY(1);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 > img {
  margin-top: 16px;
  width: 16px;
  height: 16px;
}

.c2 .transaction_item__input {
  margin-right: 40px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 .transaction_item__output {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin-left: 40px;
}

.c2 .transaction_item__output__empty {
  font-size: 16px;
  color: #666666;
  margin-top: 18px;
}

.c10 {
  width: 100%;
}

@media (min-width:750px) {
  .c4 {
    height: 20px;
  }
}

@media (max-width:750px) {
  .c4 {
    -webkit-box-pack: normal;
    -webkit-justify-content: normal;
    -ms-flex-pack: normal;
    justify-content: normal;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:750px) {
  .c4 .transaction__cell_address {
    height: 16px;
    font-size: 14px;
  }
}

@media (max-width:750px) {
  .c6 {
    margin-left: 0px;
    margin-top: 5px;
    height: 16px;
    width: 100%;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

@media (max-width:750px) {
  .c12 img {
    width: 12px;
    height: auto;
  }
}

@media (max-width:750px) {
  .c5 {
    width: 12px;
    margin: 0 10px 0 0;
  }
}

@media (max-width:750px) {
  .c11 {
    width: 12px;
    height: 12px;
    margin: 0 5px 0 7px;
  }
}

@media (max-width:750px) {
  .c7 .decimal__unit {
    margin-bottom: 0px;
  }
}

@media (max-width:1000px) {
  .c8 {
    font-size: 11px;
  }
}

@media (max-width:750px) {
  .c8 {
    margin-bottom: 0px;
  }
}

@media (max-width:1000px) {
  .c9 {
    font-size: 11px;
  }
}

@media (max-width:750px) {
  .c9 {
    margin-bottom: 0px;
  }
}

@media (max-width:750px) {
  .c3 .transaction_item__view_all {
    font-size: 14px;
    margin-top: 15px;
    height: 16px;
  }
}

@media (max-width:750px) {
  .c0 {
    padding: 15px 20px 15px 20px;
    font-size: 13px;
  }
}

@media (max-width:750px) {
  .c1 .transaction_item__content {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: normal;
    -webkit-justify-content: normal;
    -ms-flex-pack: normal;
    justify-content: normal;
    margin-bottom: 10px;
  }
}

@media (max-width:750px) {
  .c1 .transaction_item__content .transaction_item__block {
    font-weight: normal;
  }
}

@media (max-width:1200px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:1200px) {
  .c2 .transaction_item__input {
    margin: 0px;
    -webkit-flex: none;
    -ms-flex: none;
    flex: none;
    width: 100%;
  }
}

@media (max-width:1200px) {
  .c2 .transaction_item__output {
    margin: 0px;
    -webkit-flex: none;
    -ms-flex: none;
    flex: none;
    width: 100%;
  }
}

@media (max-width:750px) {
  .c2 .transaction_item__output__empty {
    font-size: 13px;
  }
}

<div
  className="c0"
  id=""
>
  <div
    className="c1"
  >
    <div
      className="transaction_item__content"
    >
      <a
        href="/transaction/0x89699abf948440ce172a6ef118cbf6986aa7c80d24d57d9309ca11929cebe5c4"
        onClick={[Function]}
      >
        <span
          className="transaction_item__hash monospace"
        >
          0x89699abf948440ce172a6ef118c...7c80d24d57d9309ca11929cebe5c4
        </span>
      </a>
      <div
        className="transaction_item__block"
      >
        (Block 1,542,388)  2020/04/23 13:34:56
      </div>
    </div>
  </div>
  <div
    className="c2"
  >
    <div
      className="transaction_item__input"
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          <div
            className="transaction__cell_address"
          >
            <a
              href="/transaction/0xbebc637dd1eee18e73590683478b6038b75427a6d3557d90e34b9a0b44c7c2cc#1"
              onClick={[Function]}
            >
              <img
                alt="left arrow"
                className="c5 transaction__cell_left_arrow"
                src="right_green_arrow.png"
              />
            </a>
            <a
              className="monospace"
              href="/address/ckb1qyqdpmfusmag3m9qjz3fenwa3qym3nc8prsqxxggjg"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              ckb1qyqdpmf...8prsqxxggjg
            </a>
          </div>
          <div
            className="c6"
          >
            <div
              className="transaction__cell__without__icon"
            >
              <div
                className="c7"
              >
                <span>
                  100
                </span>
                <div
                  className="c8 monospace"
                >
                  .2357719
                </div>
                <div
                  className="c9 monospace"
                >
                  0
                </div>
                <div
                  className="decimal__unit monospace"
                >
                  CKB
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="c4"
        >
          <div
            className="transaction__cell_address"
          >
            <a
              href="/transaction/0x7d5b5854a6d9a8dfb33188463d51d8c2d03a6b160815667cf15101f7b180f44b#75"
              onClick={[Function]}
            >
              <img
                alt="left arrow"
                className="c5 transaction__cell_left_arrow"
                src="right_green_arrow.png"
              />
            </a>
            <a
              className="monospace"
              href="/address/ckb1qyq93m8fdcx2knfmjyyn4q69zyhy7ggg835sl03lwy"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              ckb1qyq93m8...g835sl03lwy
            </a>
          </div>
          <div
            className="c6"
          >
            <div
              className="transaction__cell__without__icon"
            >
              <div
                className="c7"
              >
                <span>
                  1,902
                </span>
                <div
                  className="c8 monospace"
                >
                  .96799086
                </div>
                <div
                  className="c9 monospace"
                >
                  
                </div>
                <div
                  className="decimal__unit monospace"
                >
                  CKB
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="c4"
        >
          <div
            className="transaction__cell_address"
          >
            <a
              href="/transaction/0x49053d8717a41115a0c797a42abb3a4880ea2cdfbb8458f773b651b02737959c#62"
              onClick={[Function]}
            >
              <img
                alt="left arrow"
                className="c5 transaction__cell_left_arrow"
                src="right_green_arrow.png"
              />
            </a>
            <a
              className="monospace"
              href="/address/ckb1qyq93m8fdcx2knfmjyyn4q69zyhy7ggg835sl03lwy"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              ckb1qyq93m8...g835sl03lwy
            </a>
          </div>
          <div
            className="c6"
          >
            <div
              className="transaction__cell__without__icon"
            >
              <div
                className="c7"
              >
                <span>
                  2,002
                </span>
                <div
                  className="c8 monospace"
                >
                  .21141693
                </div>
                <div
                  className="c9 monospace"
                >
                  
                </div>
                <div
                  className="decimal__unit monospace"
                >
                  CKB
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="c4"
        >
          <div
            className="transaction__cell_address"
          >
            <a
              href="/transaction/0x59d72c3e3a5412a5e90eb1466eb801cdfaa6ccfa180d84a594c88fd426945c11#79"
              onClick={[Function]}
            >
              <img
                alt="left arrow"
                className="c5 transaction__cell_left_arrow"
                src="right_green_arrow.png"
              />
            </a>
            <a
              className="monospace"
              href="/address/ckb1qyq93m8fdcx2knfmjyyn4q69zyhy7ggg835sl03lwy"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              ckb1qyq93m8...g835sl03lwy
            </a>
          </div>
          <div
            className="c6"
          >
            <div
              className="transaction__cell__without__icon"
            >
              <div
                className="c7"
              >
                <span>
                  2,039
                </span>
                <div
                  className="c8 monospace"
                >
                  .6662154
                </div>
                <div
                  className="c9 monospace"
                >
                  0
                </div>
                <div
                  className="decimal__unit monospace"
                >
                  CKB
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <img
      alt="input and output"
      src="input_arrow_output_down.png"
    />
    <div
      className="transaction_item__output"
    >
      <div
        className="c3"
      >
        <div
          className="c10"
        >
          <div
            className="c4"
          >
            <div
              className="transaction__cell_address"
            >
              <a
                className="monospace"
                href="/address/ckb1qyqwr7vnfh6pcw2az25kkz3f4eq5a7ggec3qjyrfah"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                ckb1qyqwr7v...gec3qjyrfah
              </a>
              <img
                alt="right arrow"
                className="c11"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                src="live_cell.png"
              />
            </div>
            <div
              className="c6"
            >
              <div
                className="c12"
              >
                <div
                  className="c7"
                >
                  <span>
                    5,900
                  </span>
                  <div
                    className="c8 monospace"
                  >
                    
                  </div>
                  <div
                    className="c9 monospace"
                  >
                    .00000000
                  </div>
                  <div
                    className="decimal__unit monospace"
                  >
                    CKB
                  </div>
                </div>
                <img
                  alt="withdraw"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  src="nervos_dao_cell.png"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="c10"
        >
          <div
            className="c4"
          >
            <div
              className="transaction__cell_address"
            >
              <a
                className="monospace"
                href="/address/ckb1qyq907p79d6kkz7kxgu0awea8esvafm0586sh3kk9v"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                ckb1qyq907p...0586sh3kk9v
              </a>
              <img
                alt="right arrow"
                className="c11"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                src="live_cell.png"
              />
            </div>
            <div
              className="c6"
            >
              <div
                className="transaction__cell__without__icon"
              >
                <div
                  className="c7"
                >
                  <span>
                    145
                  </span>
                  <div
                    className="c8 monospace"
                  >
                    .08134691
                  </div>
                  <div
                    className="c9 monospace"
                  >
                    
                  </div>
                  <div
                    className="decimal__unit monospace"
                  >
                    CKB
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
